app/src/main/java/org/oppia/app/player/state/StateFragment.kt

```diff
@@ -5,22 +5,35 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.lifecycle.Observer
 import org.oppia.app.fragment.InjectableFragment
+import org.oppia.app.model.CellularDataPreference
 import org.oppia.app.player.audio.CellularDataDialogFragment
 import org.oppia.app.player.audio.CellularDataInterface
+import org.oppia.domain.audio.CellularDialogController
+import org.oppia.util.data.AsyncResult
 import javax.inject.Inject
 
 private const val TAG_CELLULAR_DATA_DIALOG = "CELLULAR_DATA_DIALOG"
 
 /** Fragment that represents the current state of an exploration. */
-class StateFragment : InjectableFragment(), CellularDataInterface {
+class StateFragment @Inject constructor(
+  private val cellularDialogController: CellularDialogController
+) : InjectableFragment(), CellularDataInterface {
   @Inject
   lateinit var stateFragmentPresenter: StateFragmentPresenter
-  // Control this boolean value from controllers in domain module.
   private var showCellularDataDialog = true
+  private var useCellularData = false
 
   init {
-    // TODO(#116): Code to control the value of showCellularDataDialog using AudioController.
+    cellularDialogController.getCellularDataPreference()
+      .observe(this, Observer<AsyncResult<CellularDataPreference>>{
+      if (it.isSuccess()) {
```

Comment:
I think it would also be nice to add test cases for StateFragment if possible to ensure the dialog is shown at the right times, and that the persistent setting works as expected.